src/main/scala/com/cognite/sdk/scala/v1/fdm/instances/InstancePropertyValue.scala

```diff
@@ -201,10 +184,22 @@ object InstancePropertyValue {
                     InstancePropertyValue.Float64List(arr.flatMap(_.asNumber).map(_.toDouble))
                 }
                 Right[DecodingFailure, InstancePropertyValue](matchingPropType)
-              case element if element.isObject =>
-                Right[DecodingFailure, InstancePropertyValue](
-                  InstancePropertyValue.ObjectList(arr)
-                )
+              case element if element.isObject && isDirectRelation(element) =>
```

Comment:
also on current setup but the code is looking at first element to decide the type, wonder if in some cases it makes it accept wrong types for 2nd+ element :thinking:

Reply:
In the case of direct relation there should be a helpful error message since I collect all the errors that I get trying to deserialize the rest of the array into this list of lefts and then make a message out of it, but I can't promise it's better everywhere.